addons/mail/models/res_users.py

```diff
@@ -148,6 +181,67 @@ def unlink(self):
         self._unsubscribe_from_non_public_channels()
         return super().unlink()
 
+    def _notify_security_setting_update(self, subject, content, mail_values=None, **kwargs):
+        """ This method is meant to be called whenever a sensitive update is done on the user's account.
+        It will send an email to the concerned user warning him about this change and making some security suggestions.
+
+        :param str subject: The subject of the sent email (e.g: 'Security Update: Password Changed')
+        :param str content: The text to embed within the email template (e.g: 'Your password has been changed')
+        :param kwargs: 'suggest_password_reset' key:
+            Whether or not to suggest the end-user to reset
+            his password in the email sent.
+            Defaults to True. """
+
+        mail_create_values = []
+        for user in self:
+            body_html = self.env['ir.qweb']._render(
+                'mail.account_security_setting_update',
+                user._notify_security_setting_update_prepare_values(content, **kwargs),
+                minimal_qcontext=True,
+            )
+
+            body_html = self.env['mail.render.mixin']._render_encapsulate(
+                'mail.mail_notification_light',
+                body_html,
+                add_context={
+                    # the 'mail_notification_light' expects a mail.message 'message' context, let's give it one
+                    'message': self.env['mail.message'].sudo().new(dict(body=body_html, record_name=user.name)),
+                    'model_description': _('Account'),
+                    'company': user.company_id,
+                },
+            )
+
+            vals = {
+                'auto_delete': True,
+                'body_html': body_html,
+                'email_from': (
+                    user.company_id.partner_id.email_formatted or
```

Comment:
Could probably compute an author (partner), and set author_id and email_from based on this partner.